### DISCUSSION TITLE
mc mirror copies the files as directory

### DISCUSSION THREAD

**Discussion by iammuho** [minio/minio#20573] [Q&A]
Hello,

I'm trying to copy my files from one minio to another in the kubernetes - pods are running at that time. I'm using below command:

```
./mc.RELEASE.2020-01-03T20-33-14Z mirror source/minio target/minio
```

and once the files are succesfuly uploaded, I can see the `xl.json` and `part.1` becomes another folder which they have their own `part.1` and `xl.json`

example:

source:/storage/abc.png/xl.json -> target:/storage/abc.png/xl.json/xl.json

how can I migrate my files ? I use 4 replica distro mode and can I use the RSYNC or any solution with the MC ?

Update: 

I've rsynced  the data to the new server disks (4 of them). In the console, I can see the objects and they are correctly formed as xl.json and part.1 but now the issue is in mattermost, some of them are missing and some are half-sized. Why that can be ?

**Comment by klauspost** [comment]
First of all, don't use an ancient version of mc.

Second of all, you must use the API to copy. Sounds like you are copying backend files. You are *never* supposed to touch backend files.

**Reply by iammuho** [reply]
Hey Klaus,

I would love to upgrade but because of mattermost, it's really old version. How can i use the API to copy ? My old minio doesn't work right now, it's terminated and have just files, I mean I can run minio server on that files, but there are 4 different disks - as it was distributed mode. 

Is there any docs I can read about that API to copy ?  As you see, it's copying files as directories.. somehow

Minio mirror/cp doesn't work.

**Reply by iammuho** [reply]
Hey @klauspost ,

Is there any way I can find my old access/secret keys or rotate them without knowing ? The old cluster has been completely destroyed and no way to get secrets but disks are there. I checked the .minio_sys/config.json/part.1 and xl.json but nothing there. Any way to read these files to retrieve secrets for root ?
